hail/src/main/scala/is/hail/io/CodecSpec.scala

```diff
@@ -24,6 +24,8 @@ trait AbstractTypedCodecSpec extends Spec {
 
   def decodedPType(requestedType: Type): PType
 
+  def decodedPType(): PType = encodedType.decodedPType(encodedVirtualType)
```

Comment:
moved stuff up from the concrete TypedCodecSpec implementation so it's on the abstract interface, which TableLiteral uses.